### DISCUSSION TITLE
Suggested feature: utility functions

### DISCUSSION THREAD

**Discussion by rChaoz** [skeletonlabs/skeleton#2620] [Ideas]
I've been using Skeleton and it's been amazing, but I feel like it could use a couple more utility functions, not necessarily UI-related. For example, Mantine for React is [full](https://mantine.dev/hooks/use-click-outside/) of such utilities. The only such utility currently in Skeleton is the local storage store, which I find very useful.

That being said, I wanted to ask if there are plans to add such utilities to Skeleton Next and, if yes, how/where to contribute.

Currently, I have a couple utility functions I am using in my project, but a debounced value function I think would fit very well. My implementation is `debounced(value, delay)` that returns a pair of stores `[denounced, instant]`. Both are normal stores - however, changes to `instant` are forwarded to `debounded` after a delay. Manual changes to `debounced` are instantly seen in both stores. This covers all use cases I can think of - including the ability to "hard" set a value.

If all is good, I was thinking of porting this function + tests to Svelte 5 with state runes and contribute to Skeleton, and maybe think of other such functions that would fit.

**Comment by endigo9740** [comment]
Hey @rChaoz, utility functions will likely be outside the scope and direction Skeleton is heading. However, some of our friends and fellow Skeleton contributors are working on this. Which I think is exactly what you're looking for!

https://github.com/svecosystem/runed

It's early days, as most things Svelte 5 related are, but I have full confidence in the folks building this! And I could see this pairing with Skeleton in a really meaningful way.
